### PR TITLE
feat(CTCP-3847): add customs office identifier controller to ie170 journey

### PR DESCRIPTION
**Pull request by malaw-hmrc**

### CODE REVIEW

File: app/controllers/IndexController.scala [hunk 1]
```diff
@@ -48,16 +49,17 @@ class IndexController @Inject() (
             lrn =>
               sessionRepository
                 .set(
-                  request.userAnswers.getOrElse(
-                    UserAnswers(
+                  request.userAnswers.getOrElse {
+                    val userAnswers = UserAnswers(
```
Comment (jackheslop96) [inline app/controllers/IndexController.scala:53]: Minor refactor suggestion:

```
request.userAnswers
    .flatMap {
        _.set(CustomsOfficeOfDeparturePage, departureData.data.CustomsOfficeOfDeparture).toOption
    }
    .getOrElse {
        UserAnswers(
            departureId,
            request.eoriNumber,
            lrn.value,
            JsObject.empty,
            timeMachine.now(),
            departureData.data
        )
    }
```

Comment (malaw-hmrc) [reply app/controllers/IndexController.scala]: Done.

File: app/models/CustomsOfficeIdentifier.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json.{Format, Json}
+
+case class CustomsOfficeIdentifier(`type`: String, description: String) extends Radioable[CustomsOfficeIdentifier] {
```
Comment (jackheslop96) [inline app/models/CustomsOfficeIdentifier.scala:21]: Is this class being used?

File: conf/messages [hunk 3]
```diff
@@ -202,3 +202,11 @@ transitOperation.addArrivalDateYesNo.checkYourAnswersLabel = Do you want to add
 transitOperation.addArrivalDateYesNo.error.required = Select yes if you want to add the arrival date at the office of destination
 transitOperation.addArrivalDateYesNo.change.hidden = if you want to add the arrival date at the office of destination
 
+
+locationOfGoods.customsOfficeIdentifier.title = What is the customs office identifier for the location of goods?
+locationOfGoods.customsOfficeIdentifier.heading = What is the customs office identifier for the location of goods?
+locationOfGoods.customsOfficeIdentifier.hint = Enter the office location or code, like Dover or GB000060.
+locationOfGoods.customsOfficeIdentifier.placeholder = Select a customsOffice
```
Comment (jackheslop96) [inline conf/messages:209]: `Select a customs office identifier`

Comment (malaw-hmrc) [reply conf/messages]: Updated... Done

File: app/models/CustomsOfficeIdentifier.scala (hunk 2 above)
Comment (malaw-hmrc) [reply app/models/CustomsOfficeIdentifier.scala]: Oops removed...

**Review by jackheslop96 — DISMISSED** [review]

**Review by rajukakumanu86 — APPROVED** [review]

**Pull request merged**